### DISCUSSION TITLE
[Migration] Create table with PostgreSQL schema name

### DISCUSSION THREAD

**Discussion by 23prime** [SeaQL/sea-orm#1029] [Q&A] [answered]
I want to use `sea-orm-cli` `v0.9.2` for DB migration.
And I use PostgreSQL.

Now, I have create an migration file, and run:

```console
$ sea-orm-cli migrate up
```

A table is created in the default schema `public`.

But, I want to create the table to non `public` schema.
How I can?

I tried these, but they didn't work as expected:

- Implement `Iden` manually like `unquoted(Table) => "schema_name.table_name`. Then `schema_name.table_name` table were created in `public` schema.
- Run with an option `-s schema_name` like [Generating Entity Files](https://www.sea-ql.org/SeaORM/docs/generate-entity/sea-orm-cli/#generating-entity-files).
- Set an env variable `DATABASE_SCHEMA=schema_name` and run.

**Comment by 23prime** [comment, marked as answer]
I solved it myself.

1. Define `Schema` and `Table` as different items.
2. Implement `Iden` manually.
3. Give both `Schema` and `Table` to `table` method as a tuple.

```rs
enum MyEntity {
    Schema,
    Table,
    Id,
    // more columns...
}
```

```rs
impl Iden for MyEntity {
    fn unquoted(&self, s: &mut dyn std::fmt::Write) {
        write!(
            s,
            "{}",
            match self {
                Self::Schema => "schema_name",
                Self::Table => "table_name",
                Self::Id => "id",
                // more columns...
            }
        )
        .unwrap();
    }
}
```

```rs
impl MigrationTrait for Migration {
    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
        manager
            .create_table(
                Table::create()
                    .table((MyEntity::Schema, MyEntity::Table)) // here
                    .if_not_exists()
                    .col(ColumnDef::new(MyEntity::Id).integer().not_null().auto_increment().primary_key())
                    // more columns...
                    .to_owned(),
            )
            .await
    }

    async fn down(&self, manager: &SchemaManager) -> Result<(), DbErr> {
        manager
            .drop_table(
                Table::drop()
                    .table((MyEntity::Schema, MyEntity::Table)) // here
                    .to_owned()
            )
            .await
    }
}
```

Thank you.

## References

- https://github.com/SeaQL/sea-query/blob/cfaa483ff060051241c495a822cb05cccfd95bcd/src/table/create.rs#L133
- https://github.com/SeaQL/sea-query/blob/cfaa483ff060051241c495a822cb05cccfd95bcd/src/types.rs#L321

**Reply by billy1624** [reply]
Hey @23prime, yes, this is correct! Sorry for the delay.
